contrib/samples/openvino/deployer/component.yaml

```diff
@@ -0,0 +1,107 @@
+apiVersion: argoproj.io/v1alpha1
```

Comment:
The `component.yaml` format is a bit different. It's much smaller. For this component it should be:

```
name: OpenVINO model server deployer
description: Triggers deployment of [OpenVINO Model Server](https://github.com/IntelAI/OpenVINO-model-server) to Kubernetes.
inputs:
- name: Batch size
  default: auto
- name: Log level
  default: DEBUG
- name: Model export path
  default: gs://intelai_public_models/resnet_50_i8
- name: Model version policy
  default: '{"latest": { "num_versions":2 }}'
- name: Replicas
  default: '1'
- name: Server name
  default: resnet
outputs:
- name: Server endpoint
implementation:
  container:
    image: gcr.io/constant-cubist-173123/inference_server/ml_deployer:9
    command: [./deploy.sh]
    args: [
      --model-export-path, {inputValue: Model export path},
      --server-name, {inputValue: Server name},
      --log-level, {inputValue: Log level},
      --batch-size, {inputValue: Batch size},
      --model-version-policy, {inputValue: Model version policy},
      --replicas, {inputValue: Replicas},
      --server-endpoint-output-file, {outputPath: Server endpoint},
    ]
```

Reply:
Other than this,
/lgtm

Reply:
how do you generate this format of yaml file? dsl-compile creates more verbose form.

Reply:
>dsl-compile creates more verbose form.

dsl-compiler compiles a full pipeline, a graph that possibly consists of many operations, not a single component. Also the Argo Workflow format is quite verbose.

>how do you generate this format of yaml file

Currently there are two ways:
* Create by hand: Copy some existing `component.yaml` (e.g. [Keras - Train classifier](https://github.com/kubeflow/pipelines/blob/master/components/sample/keras/train_classifier/component.yaml) or [Bigquery - Query](https://github.com/kubeflow/pipelines/blob/master/components/gcp/bigquery/query/component.yaml)), look at your command-line arguments and modify accordingly. I usually use this method.
* Create an instance of the `ComponentSpec` class and then serialize it to file.

```
from kfp.components._structures import *
#Build component spec using strongly-typed Python structures (with auto-complete, validation etc)
component_spec = ComponentSpec(
   name='Predict using TF on Dataflow',
   description='Runs TensorFlow prediction on Google Cloud Dataflow',
   inputs=[
       InputSpec(name='Data file pattern',   type='GcsUri',        description='GCS of test file patterns.'},
       InputSpec(name='Schema',              type='GcsUri',       description='GCS json schema file path.'},
       InputSpec(name='Target',              type='String',                 description='Name of the column for prediction target.'},
       InputSpec(name='Model',               type='GcsUri', description='GCS of exported trained models.'},
       InputSpec(name='Run mode',            type='String', description='Whether to run the job locally or in Cloud Dataflow.'},
       InputSpec(name='GCP project',         type='GCP project',            description='The GCP project to run the dataflow job.'},
   ],
   outputs=[
       OutputSpec(name='Predictions dir',    type='GcsUri',  description='GCS path to store model predictions.')
   ],
   implementation=ImplementationSpec(
       container=ContainerSpec(
           image='gcr.io/...',
           command=['python3', '/kfp/src/predict.py'],
           args=[
               '--data',    InputValueArgument('Data file pattern'),
               '--schema',  InputValueArgument('Schema'),
               '--target',  InputValueArgument('Target'),
               '--model',   InputValueArgument('Model'),
               '--mode',    InputValueArgument('Run mode'),
               '--project', InputValueArgument('GCP project'),
               '--output',  OutputPathArgument('Predictions dir'),
           ],
       )
   )
)
#Write component artifact file
with open('component.yaml', 'w') as f:
   yaml.dump(component_spec.to_struct(), f)
```

You can validate the `component.yaml` text by calling `kfp.component.load_component_from_text(...)` or validate the file using `kfp.component.load_component_from_file('component.yaml')`.

How would you like us to improve the component authoring process?

P.S. The `component.yaml` I've posted in the previous comment is the correct specification for your component, so you should probably just take it.